helm-chart/binderhub/templates/deployment.yaml

```
@@ -102,8 +102,12 @@ spec:
           value: {{ .Values.jupyterhub.hub.services.binder.oauth_client_id | quote}}
         - name: JUPYTERHUB_OAUTH_CALLBACK_URL
           value: {{ .Values.jupyterhub.hub.services.binder.oauth_redirect_uri | quote}}
+        {{- if .Values.jupyterhub.hub.allowNamedServers }}
         - name: JUPYTERHUB_ALLOW_NAMED_SERVERS
-          value: {{ .Values.jupyterhub.hub.allowNamedServers | quote }}
+          value: true
```

```suggestion
          value: "true"
```
Environment variables must always be strings so `true` needs to be quoted, otherwise you get errors like:
```
Error: UPGRADE FAILED: cannot patch "binder" with kind Deployment: v1.Deployment.Spec: v1.DeploymentSpec.Template: v1.PodTemplateSpec.Spec: v1.PodSpec.Containers: []v1.Container: v1.Container.Env: []v1.EnvVar: v1.EnvVar.Value: ReadString: expects " or n, but found t, error found in #10 byte of ...|,"value":true},{"nam|..., bigger context ...|{"name":"JUPYTERHUB_ALLOW_NAMED_SERVERS","value":true},{"name":"JUPYTERHUB_NAMED_SERVER_LIMIT_PER_US|...
```
FYI I'm using Helm 3, does it work on Helm 2?

---

no, it doesn't work with Helm 2 neither